resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/ServerResponseWriter.java

```diff
@@ -336,12 +339,24 @@ public static void commitHeaders(BuiltResponse jaxrsResponse, HttpResponse respo
    {
       double q = -1;
       double qs = -1;
+      Class<?> writerType = null;
       
       public SortableMediaType(MediaType m)
       {
          this(m.getType(), m.getSubtype(), m.getParameters());
       }
       
+      public SortableMediaType(MediaType m, Class<?> writerType)
+      {
+         this(m.getType(), m.getSubtype(), m.getParameters(), writerType);
+      }
+      
+      public SortableMediaType(String type, String subtype, Map<String, String> parameters, Class<?> writerType)
+      {
+         this(type, subtype, parameters);
+         this.writerType = writerType;
+      }
+      
       public SortableMediaType(String type, String subtype, Map<String, String> parameters)
       {
          super(type, subtype, parameters);

```

Comment:
About the `compareTo()` method:
- Lets consider media types "application/xml" and "text/*". According to the spec "application/xml" > "text/*" since n/m > n/* > */* . Even if media types are not compatible the comparaison should not be based on "qs" nor "q" since n/m > n/* > */*.